lib/sessionParticipants/index.js

```diff
@@ -20,16 +24,18 @@ function matchingIndex(issuer, sessionIndex, nameId){
  * used to find the correct Session Participant object which
  * represents the issuer of the previous mentions request/response.
  * 
- * @issuer {string}   The string as it was received in the SAML request/response
- * @sessionIndex {string}   The string as it was received in the SAML request/response. Only available in LogoutRequests
- * @cb     {function} The callback that will be called with '(err, sessionParticipant)'
+ * @issuer {string} The string as it was received in the SAML request/response
+ * @sessionIndices {string|string[]} An single string or array of strings representing the
+ * SessionIndex values as received in the SAML request/response. Only available in LogoutRequests.
+ * @nameId {string} The value of the nameId element from the SAML request/response
+ * @cb {function} The callback that will be called with '(err, sessionParticipant)'
  */
-SessionParticipants.prototype.get = function(issuer, sessionIndex, nameId, cb) {
+SessionParticipants.prototype.get = function(issuer, sessionIndices, nameId, cb) {
```

Comment:
Isn't it the case that multiple sessions/participants may match? Should this not be returning a collection of session participants now?

Reply:
I've been mulling over this one and I don't think it would make sense to return multiple session participants. Let me try to share my reasoning and interpretations from the spec.

[Sec 3.7.3.2 Session Authority Rules](http://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf) states:

```
Terminate the principal's current session as specified by the <saml:BaseID>, <saml:NameID>, or <saml:EncryptedID> element, and any <SessionIndex> elements present in the logout request message.
```

When a user's current session is looked up at the IdP when initiating a SLO from an SP, the IdP should generate a list of all session participants and begin the SLO flow. During the SLO flow, the IdP will ask each session participant to log the user out before finally terminating its own authority session.

A session participant in this case would be any SP that has authenticated with that user (under their current session) with the IdP.

It could be that the same SP has authenticated and stored the SessionIndex resulting from the authentication statement from the IdP for the same user several times (perhaps due to different session timeout lengths or other reasons). However, the IdP should still see that as a single session participant for that user's current session.

> Should this not be returning a collection of session participants now?

To answer this more directly based on my understanding: within the context of a user's current session at the IdP, the SP requesting the SLO (via the LogoutRequest) may specify multiple SessionIndex elements but it should only map to a single session participant.

It's like the SP is saying to the IdP: "hey, I have these session indices for this user that you authenticated. I'm not sure which indices you have on file, but if any of these match, log this user out"

The IdP will then find any of those session indices that match for that SP and then loop through all it's session participants with SLO enabled and log them out before clearing the authority's session (at the IdP itself).

Does my reasoning/interpretation make sense?